### PR TITLE
Addition of a responsive toolbar

### PR DESCRIPTION
**Pull request by elpoelma**
This PR introduces the `ResponsiveToolbar` component, a modified version of the current `Toolbar` component which takes screen and container widths into account. 
The new toolbar component consists of two main sections:
- A `main` section which will typically contain often used controls such as text styling controls, table controls, history controls etc.
- A `side` section which can include additional controls. It will typically contain additional information and formatting controls.

Both of these sections are collapsible. If the controls of a section no longer fit in the available space, they are moved to a collapsible dropdown. The toolbar will first try to collapse the `side` section, and afterwards the `main` section

The component should be used in the following manner:

```
<ResponsiveToolbar>
<:main>
{{!-- A list of main controls --}}
</:main>
<:side>
{{!-- A list of side controls --}}
</:side>
<ResponsiveToolbar>
```

The toolbar will collapse each direct child of the `main` and `side` sections separately. If you do not wish a specific child to be collapsed, you may add the `data-ignore-resize` attribute.

This PR is a solution to https://binnenland.atlassian.net/browse/GN-4120?atlOrigin=eyJpIjoiYzk4MDcxZmNhZTJjNDJiNjkxYjQwOTM4YTNmYjFiNmIiLCJwIjoiaiJ9.

### CODE REVIEW

**Review by lagartoverde — APPROVED** [review]
Looks good to me :)

File: addon/components/responsive-toolbar.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+  @action
+  toggleSideDropdown() {
+    this.side.showDropdown = !this.side.showDropdown;
+    if (this.side.showDropdown) {
+      this.main.showDropdown = false;
+    }
+  }
+
+  isOverflowing = () => {
```
Comment (abeforgit) [inline addon/components/responsive-toolbar.ts:90]: any reason this is not a getter? (if there is, would like a code comment explaining why)

File: addon/components/responsive-toolbar.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { modifier } from 'ember-modifier';
+import { tracked } from 'tracked-built-ins';
+
+export default class ResponsiveToolbar extends Component {
+  toolbar?: HTMLElement;
+
+  main: {
```
Comment (abeforgit) [inline addon/components/responsive-toolbar.ts:9]: would like to see a named interface here

**Review by abeforgit — CHANGES_REQUESTED** [review]
works really nicely, good work. Comments are nitpicks. Feel free to merge after adressing

File: addon/components/responsive-toolbar.ts (hunk 1 above)
Comment (elpoelma) [reply addon/components/responsive-toolbar.ts]: Not really :stuck_out_tongue:

**Pull request merged**